### PR TITLE
state: move variables tests to use must library.

### PR DESCRIPTION
**Pull request by jrasell**
Small piece of admin while starting locking work.

### CODE REVIEW

**Review by tgross — APPROVED** [review]
LGTM!

File: nomad/state/state_store_variables_test.go [hunk 1]
```diff
@@ -75,17 +75,17 @@ func TestStateStore_UpsertVariables(t *testing.T) {
...
 
 		// Ensure the create and modify indexes are populated correctly.
-		require.Equal(t, uint64(21), got[0].CreateIndex, "%s: incorrect create index", got[0].Path)
-		require.Equal(t, uint64(21), got[0].ModifyIndex, "%s: incorrect modify index", got[0].Path)
-		require.Equal(t, uint64(22), got[1].CreateIndex, "%s: incorrect create index", got[1].Path)
-		require.Equal(t, uint64(22), got[1].ModifyIndex, "%s: incorrect modify index", got[1].Path)
+		must.Eq(t, uint64(21), got[0].CreateIndex, must.Sprintf("%s: incorrect create index", got[0].Path))
+		must.Eq(t, uint64(21), got[0].ModifyIndex, must.Sprintf("%s: incorrect modify index", got[0].Path))
+		must.Eq(t, uint64(22), got[1].CreateIndex, must.Sprintf("%s: incorrect create index", got[1].Path))
+		must.Eq(t, uint64(22), got[1].ModifyIndex, must.Sprintf("%s: incorrect modify index", got[1].Path))
```
Comment (shoenig) [inline nomad/state/state_store_variables_test.go:84]: I suspect these uint casts can be eliminated

File: nomad/state/state_store_variables_test.go [hunk 2]
```diff
@@ -111,16 +111,16 @@ func TestStateStore_UpsertVariables(t *testing.T) {
...
-		require.NoError(t, err)
-		require.Equal(t, insertIndex, reInsertActualIndex, "index should not have changed")
+		must.NoError(t, err)
+		must.Eq(t, insertIndex, reInsertActualIndex, must.Sprintf("index should not have changed"))
 
 		quotaUsed, err := testState.VariablesQuotaByNamespace(ws, structs.DefaultNamespace)
-		require.NoError(t, err)
-		require.Equal(t, int64(expectedQuotaSize), quotaUsed.Size)
+		must.NoError(t, err)
+		must.Eq(t, int64(expectedQuotaSize), quotaUsed.Size)
```
Comment (shoenig) [inline nomad/state/state_store_variables_test.go:123]: same?

File: nomad/state/state_store_variables_test.go [hunk 3]
```diff
@@ -161,17 +161,17 @@ func TestStateStore_UpsertVariables(t *testing.T) {
...
 
 		// update the mocks so the test element has the correct create/modify
 		// indexes and times now that we have validated them
 		svs = got
 
 		quotaUsed, err := testState.VariablesQuotaByNamespace(ws, structs.DefaultNamespace)
-		require.NoError(t, err)
-		require.Equal(t, int64(expectedQuotaSize+1), quotaUsed.Size)
+		must.NoError(t, err)
+		must.Eq(t, int64(expectedQuotaSize+1), quotaUsed.Size)
```
Comment (shoenig) [inline nomad/state/state_store_variables_test.go:174]: same?

File: nomad/state/state_store_variables_test.go [hunk 4]
```diff
@@ -310,29 +310,29 @@ func TestStateStore_DeleteVariable(t *testing.T) {
...
 			delete2Count++
 		}
-		require.Equal(t, 0, delete2Count, "unexpected number of variables in table")
+		must.Eq(t, 0, delete2Count, must.Sprintf("unexpected number of variables in table"))
 
 		quotaUsed, err := testState.VariablesQuotaByNamespace(ws, structs.DefaultNamespace)
-		require.NoError(t, err)
-		require.Equal(t, int64(0), quotaUsed.Size)
+		must.NoError(t, err)
+		must.Eq(t, int64(0), quotaUsed.Size)
```
Comment (shoenig) [inline nomad/state/state_store_variables_test.go:335]: same?

**Review by shoenig — APPROVED** [review]
LGTM; just some casts I don't think we need anymore

**Pull request merged**